src/include/duckdb/main/query_result.hpp

```diff
@@ -43,11 +44,11 @@ class BaseQueryResult {
 	//! Whether or not execution was successful
 	bool success;
 	//! The error string (in case execution was not successful)
-	string error;
+	PreservedError error;
```

Comment:
If you are reworking the way the `error` message is read anyway, I would prefer making both this and `success` private (or protected) members. In addition, perhaps leave `GetError` as-is, and add an extra `GetExceptionType` message?

Reply:
Ah yes good idea, will do 👍

Reply:
I'm adding a `DUCKDB_API const PreservedError& GetErrorObject();` for where I'm passing on the error

Reply:
Maybe it's better to have a `ThrowException` method on the QueryResult instead
GetErrorObject is only really used in combination with ToException() on the query result, apart from some rare cases where they are compared or passed on